comp/core/status/statusimpl/common_header_provider_test.go

```diff
@@ -102,6 +103,32 @@ func TestCommonHeaderProviderText(t *testing.T) {
 	assert.Equal(t, expectedResult, output)
 }
 
+func TestCommonHeaderProviderTime(t *testing.T) {
+	// test that the time is updated on each call
+	counter := 0
+	nowFunc = func() time.Time {
+		counter++
+		return time.Unix(int64(counter), 0)
+	}
+	defer func() { nowFunc = time.Now }()
+
+	config := fxutil.Test[config.Component](t, config.MockModule())
+
+	provider := newCommonHeaderProvider(agentParams, config)
+
+	data := map[string]interface{}{}
```

Comment:
why use `data := map[string]interface{}{}` and not `var data map[string]interface{}` ?

Reply:
The map needs to be initialized (allocated), if we use `var data map[string]interface{}` then it would be `nil`, and assigning to it would fail

Reply:
okay thx :D